dask/dataframe/tests/test_dataframe.py

```diff
@@ -2632,3 +2632,14 @@ def test_boundary_slice_same(index, left, right):
     df = pd.DataFrame({"A": range(len(index))}, index=index)
     result = boundary_slice(df, left, right)
     tm.assert_frame_equal(result, df)
+
+
+def test_meaningful_error_on_wrong_meta():
```

Comment:
This is a test for a specific issue, but the real behavior we'd like to test is that errors for things like `mean` on object series are more informative. Perhaps this would be better:

```python
def test_better_errors_object_reductions():
    # GH2452
    s = pd.Series(['a', 'b', 'c', 'd'])
    ds = dd.from_pandas(s, npartitions=2)
    with pytest.raises(ValueError) as err:
        ds.mean()
    assert err.value.message == "`mean` not supported with object series"
```